lib/compress.js

```diff
@@ -2670,9 +2670,15 @@ merge(Compressor.prototype, {
         }
         // x && (y && z)  ==>  x && y && z
         // x || (y || z)  ==>  x || y || z
+        // x + ("y" + z)  ==>  x + "y" + z
+        // "x" + (y + "z")==>  "x" + y + "z"
         if (self.right instanceof AST_Binary
             && self.right.operator == self.operator
-            && (self.operator == "&&" || self.operator == "||"))
+            && (self.operator == "&&" || self.operator == "||"
+            || (self.operator == "+"
+            && (self.right.left.is_string(compressor)
+            || (self.left.is_string(compressor)
+            && self.right.right.is_string(compressor))))))
```

Comment:
@kzc this practical joke has gotten seriously out of hand... any suggestions on improving readability would be greatly appreciated.